src/main/kotlin/com/cognifide/gradle/aem/pkg/deploy/SyncTask.kt

```diff
@@ -11,15 +11,24 @@ abstract class SyncTask : AemDefaultTask() {
     protected val propertyParser = PropertyParser(project)
 
     protected fun synchronizeInstances(synchronizer: (InstanceSync) -> Unit) {
-        synchronizeInstances(synchronizer, filterInstances())
+        synchronizeInstances(synchronizer, Instance.filter(project))
     }
 
     protected fun <T : Instance> synchronizeInstances(synchronizer: (InstanceSync) -> Unit, instances: List<T>) {
-        val callback = { instance: T -> synchronizeInstance(synchronizer, instance) }
-        if (config.deployParallel) {
-            instances.parallelStream().forEach(callback)
+        synchronizeInstances(instances, { instance: T -> synchronizeInstance(synchronizer, instance) })
+    }
+
+    protected fun <T : Instance> synchronizeInstances(instances: List<T>, callback: (T) -> Unit) {
+        if (instances.isEmpty()) {
+            logger.warn("No instances to synchronize. Verify deploy instance name filter '${config.deployInstanceName}' and defined instances name (environment-type).")
         } else {
-            instances.onEach(callback)
+            if (config.deployParallel) {
+                logger.info("Synchronizing ${instances.size} instance(s) in parallel mode")
```

Comment:
that log entry also helps out to understand why plugin is most often uploading and installing firstly to publish not to author (because it is faster instance and deploy is done in parallel)